Wikipedia/Code/WMFProxyServer.m

```diff
@@ -189,136 +189,122 @@ - (NSURL*)proxyURLForImageURLString:(NSString*)imageURLString {
-        regex = [NSRegularExpression regularExpressionWithPattern:pattern
-                                                          options:NSRegularExpressionCaseInsensitive
-                                                            error:nil];
+        NSString *attributePattern = @"(src|data-file-width|width)=[\"']?((?:.(?![\"']?\\s+(?:\\S+)=|[>\"']))+.)[\"']?"; //match on the three attributes we need to read: src, data-file-width, width
+        attributeRegex = [NSRegularExpression regularExpressionWithPattern:attributePattern options:NSRegularExpressionCaseInsensitive error:nil];
+        
+        NSString *sizePattern = @"^[0-9]+(?=px-)";//match on numerals followed by px- at the beginning of the filename
+        sizeRegex = [NSRegularExpression regularExpressionWithPattern:sizePattern options:NSRegularExpressionCaseInsensitive error:nil];
     });
+    
+    __block NSString *src = nil; //the image src
+    __block NSRange srcAttributeRange = NSMakeRange(NSNotFound, 0); //the range of the full src attribute - src=blah
+    __block NSInteger dataFileWidth = 0; //the original file width from data-file-width=
+    __block NSInteger width = 0; //the width of the image from width=
+    NSInteger attributeOffset = 0;
+    [attributeRegex enumerateMatchesInString:imageTagContents options:0 range:NSMakeRange(0, imageTagContents.length) usingBlock:^(NSTextCheckingResult * _Nullable attributeResult, NSMatchingFlags flags, BOOL * _Nonnull stop) {
+        NSString *attributeName = [[attributeRegex replacementStringForResult:attributeResult inString:imageTagContents offset:attributeOffset template:@"$1"] lowercaseString];
+        NSString *attributeValue = [attributeRegex replacementStringForResult:attributeResult inString:imageTagContents offset:attributeOffset template:@"$2"];
+        if ([attributeName isEqualToString:@"src"]) {
+            src = attributeValue;
+            srcAttributeRange = attributeResult.range;
+        } else if ([attributeName isEqualToString:@"data-file-width"]) {
+            dataFileWidth = [attributeValue integerValue];
+        } else if ([attributeName isEqualToString:@"width"]) {
+            width = [attributeValue integerValue];
+        }
+        *stop = dataFileWidth > 0 && srcAttributeRange.location != NSNotFound && width > 0;
+    }];
+    
+    NSMutableString *newImageTagContents = [imageTagContents mutableCopy];
+    
+    
+    NSMutableArray *srcPathComponents = [[src pathComponents] mutableCopy];
+    NSString *resizedSrc = nil;
+    //check to see if we should change the image size being requested - if we know the original file width (dataFileWidth) and the width is present and greater than or equal to 64, and it's a thumbnail URL, we should request a larger size
+    if (dataFileWidth > 0 && (width == 0 || width >= 64) && srcPathComponents.count > 4 && [[srcPathComponents[srcPathComponents.count - 5] lowercaseString] isEqualToString:@"thumb"]) {
```

Comment:
I wonder if we should check for "/thumb/" anywhere in the path instead of at a specific location? Thoughts?